src/en/allanime/src/eu/kanade/tachiyomi/extension/en/allanime/AllAnime.kt

```diff
@@ -0,0 +1,485 @@
+
+        return pages.pictureUrls.mapIndexed { index, image ->
+            Page(
+                index = index,
+                imageUrl = "$imageDomain${image.url}",
+            )
+        }
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        throw UnsupportedOperationException("Not used")
+    }
+
+    override fun imageUrlParse(response: Response): String {
+        throw UnsupportedOperationException("Not used")
+    }
+
+    private fun String.parseThumbnailUrl(): String {
+        return if (this.matches(urlRegex)) {
+            this
+        } else {
+            "$image_cdn$this?w=250"
+        }
+    }
+
+    private fun String?.parseStatus(): Int {
+        if (this == null) {
+            return SManga.UNKNOWN
+        }
+
+        return when {
+            this.contains("releasing", true) -> SManga.ONGOING
+            this.contains("finished", true) -> SManga.COMPLETED
+            else -> SManga.UNKNOWN
+        }
+    }
+
+    private fun String.titleToSlug() = this.trim()
+        .lowercase(Locale.US)
+        .replace(titleSpecialCharactersRegex, "-")
+
+    override fun setupPreferenceScreen(screen: PreferenceScreen) {
+        ListPreference(screen.context).apply {
+            key = DOMAIN_PREF
+            title = "Preferred domain"
+            entries = arrayOf("allanime.to", "allanime.co")
+            entryValues = arrayOf("allanime.to", "allanime.co")
+            setDefaultValue("allanime.to")
+            summary = "%s"
+
+            setOnPreferenceChangeListener { _, newValue ->
+                val selected = newValue as String
+                val index = findIndexOfValue(selected)
+                val entry = entryValues[index] as String
+                Toast.makeText(screen.context, RESTART_APP_STRING, Toast.LENGTH_LONG).show()
+
+                preferences.edit().putString(key, entry).commit()
+            }
+        }.let { screen.addPreference(it) }
+
+        ListPreference(screen.context).apply {
+            key = TITLE_PREF
+            title = "Preferred Title Style"
+            entries = arrayOf("Romaji", "English", "Native")
+            entryValues = arrayOf("romaji", "eng", "native")
+            setDefaultValue("romaji")
+            summary = "%s"
+
+            setOnPreferenceChangeListener { _, newValue ->
+                val selected = newValue as String
+                val index = findIndexOfValue(selected)
+                val entry = entryValues[index] as String
+
+                preferences.edit().putString(key, entry).commit()
+            }
+        }.let { screen.addPreference(it) }
+
+        SwitchPreferenceCompat(screen.context).apply {
+            key = SHOW_ADULT_PREF
+            title = "Show Adult Content"
+            setDefaultValue(false)
+
+            setOnPreferenceChangeListener { _, newValue ->
+                val showAdult = newValue as Boolean
+
+                preferences.edit().putBoolean(key, showAdult).commit()
```

Comment:
```suggestion
                preferences.edit().putBoolean(key, showAdult).commit()
```

Remove all 3 lines of `commit()` as these are unnecessary.

Reply:
I mean removing them completely. The app handles saving the values already.

Reply:
But the build fails without it.
```
Type mismatch: inferred type is Unit but Boolean was expected
```